Libraries/LibWeb/SVG/SVGElement.cpp

```diff
@@ -32,75 +33,92 @@ void SVGElement::initialize(JS::Realm& realm)
 }
 
 struct NamedPropertyID {
-    NamedPropertyID(CSS::PropertyID property_id)
+    NamedPropertyID(CSS::PropertyID property_id, Vector<FlyString> supported_elements = {})
         : id(property_id)
         , name(CSS::string_from_property_id(property_id))
+        , supported_elements(move(supported_elements))
     {
     }
 
     CSS::PropertyID id;
     StringView name;
+    Vector<FlyString> supported_elements;
 };
 
-static Array const attribute_style_properties {
-    // FIXME: The `fill` attribute and CSS `fill` property are not the same! But our support is limited enough that they are equivalent for now.
-    NamedPropertyID(CSS::PropertyID::Fill),
-    // FIXME: The `stroke` attribute and CSS `stroke` property are not the same! But our support is limited enough that they are equivalent for now.
-    NamedPropertyID(CSS::PropertyID::ClipPath),
-    NamedPropertyID(CSS::PropertyID::ClipRule),
-    NamedPropertyID(CSS::PropertyID::Color),
-    NamedPropertyID(CSS::PropertyID::Cursor),
-    NamedPropertyID(CSS::PropertyID::Direction),
-    NamedPropertyID(CSS::PropertyID::Display),
-    NamedPropertyID(CSS::PropertyID::FillOpacity),
-    NamedPropertyID(CSS::PropertyID::FillRule),
-    NamedPropertyID(CSS::PropertyID::FontFamily),
-    NamedPropertyID(CSS::PropertyID::FontSize),
-    NamedPropertyID(CSS::PropertyID::FontStyle),
-    NamedPropertyID(CSS::PropertyID::FontWeight),
-    NamedPropertyID(CSS::PropertyID::ImageRendering),
-    NamedPropertyID(CSS::PropertyID::LetterSpacing),
-    NamedPropertyID(CSS::PropertyID::Mask),
-    NamedPropertyID(CSS::PropertyID::MaskType),
-    NamedPropertyID(CSS::PropertyID::Opacity),
-    NamedPropertyID(CSS::PropertyID::Overflow),
-    NamedPropertyID(CSS::PropertyID::PointerEvents),
-    NamedPropertyID(CSS::PropertyID::StopColor),
-    NamedPropertyID(CSS::PropertyID::StopOpacity),
-    NamedPropertyID(CSS::PropertyID::Stroke),
-    NamedPropertyID(CSS::PropertyID::StrokeDasharray),
-    NamedPropertyID(CSS::PropertyID::StrokeDashoffset),
-    NamedPropertyID(CSS::PropertyID::StrokeLinecap),
-    NamedPropertyID(CSS::PropertyID::StrokeLinejoin),
-    NamedPropertyID(CSS::PropertyID::StrokeMiterlimit),
-    NamedPropertyID(CSS::PropertyID::StrokeOpacity),
-    NamedPropertyID(CSS::PropertyID::StrokeWidth),
-    NamedPropertyID(CSS::PropertyID::TextAnchor),
-    NamedPropertyID(CSS::PropertyID::TextRendering),
-    NamedPropertyID(CSS::PropertyID::TextOverflow),
-    NamedPropertyID(CSS::PropertyID::TransformOrigin),
-    NamedPropertyID(CSS::PropertyID::UnicodeBidi),
-    NamedPropertyID(CSS::PropertyID::Visibility),
-    NamedPropertyID(CSS::PropertyID::WhiteSpace),
-    NamedPropertyID(CSS::PropertyID::WordSpacing),
-    NamedPropertyID(CSS::PropertyID::WritingMode),
-};
+static Vector<NamedPropertyID> attribute_style_properties()
+{
+    static Vector const properties = {
```

Comment:
Does this need to be a Vector? Can we still store an Array and return a ReadonlySpan?
(if it does need to be a Vector for some reason, let's at least return it by const&)

Reply:
That's better. I've changed this to return a `ReadonlySpan`.